lib/client/js/ui/controls/AuthButton.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * The UI Control that manages the AuthButton button component
+ * residing in the Header section
+ *
+ * @class
+ */
+var AuthButton = (function () {
+
+    /**
+     * @alias AuthButton
+     */
+    function AuthButton(config) {
+        // Call the super
+        Control.apply(this, [config]);
+    }
+
+    AuthButton.prototype = new Control();
+
+    AuthButton.prototype.view = function () {
+        if (this.getProp("loggedInUser") === null) {
+            return $("<button>", {"class": "beans--AuthButton"})
+                .text("Login");
+        } else {
+            return $("<button>", {"class": "beans--AuthButton"})
+                .text("Logout");
+        }
+    };
+
+    AuthButton.prototype.attachEvent = function () {
+        var self = this;
+        $(".beans--AuthButton").click(function () {
+            if ($(this).text() === "Login") {
```

Comment:
I don't exactly understand what you have in mind for this change. I can change this to utilize a ternary operator like you suggested above. But, I'm unsure of how to get away from comparing against either the "login" or "logout" strings. Since neither one is null, we wouldn't be able to just check for truthiness, right? Please elaborate.